added json files to support easier downloading

wkelly17 opened the pull request:
This commit creates a download.json file type for the entire repository, and it creates book level "whole.json" files in each book.  
 - Bible
 - Commentary
 - notes
 - questions

This allows for downloading an entire repository into memory, such that it could be cached in teh browser or a service worker etc;  The main advantage to this json format instead of just a whole.html at the book or repository level is that the individual responses can be cached or accessed more granularly.   E.g, the entire bible can be requested in 1 request to blob storage vs 1000+ if desired. 

Edit: @PurpleGuitar .. just in case you didn't see this as well.

wkelly17 commented:
Also, the diff is really muddy cause it looks like we have competing formatters @rbnswartz ... If you have some suggestion or request to clean it up, let me know.

rbnswartz commented:
@wkelly17 Looks like you use tabs instead of spaces, just replace the tabs with spaces and the diff should match up better.

rbnswartz reviewed (commented):
The implementation is pretty solid, I have a couple comments on variable names. We're missing bible output from this so we should put that in.

PurpleGuitar commented on ScriptureRenderingPipeline/Renderers/BibleRenderer.cs:
```diff
@@ -16,152 +16,152 @@
+        var tableOfContentsMarkers = tmp.GetChildMarkers<TOC3Marker>();
+        if (tableOfContentsMarkers.Count == 0)
+        {
+          var bookAbbrivation = GetBookAbberviationFromFileName(f);
```
Typo: Should be "bookAbbreviation" -- this typo appears in a lot of places in this file.

PurpleGuitar commented on ScriptureRenderingPipeline/Renderers/ScripturalMarkdownRendererBase.cs:
```diff
@@ -14,196 +14,219 @@
+          Label = book.BookName,
+          Slug = book.BookId
+        };
+        var outPutBookDataIncluded = new OutputBook()
```
typo: should be "outputBookDataIncluded".

Also, I'd prefer a different term than "Data" -- what is this data?  Rendered content?  Maybe it should be "outputBookContentIncluded"?

wkelly17 commented:
@rbnswartz  @PurpleGuitar 
[4b86fef](https://github.com/WycliffeAssociates/ScriptureRenderingPipeline/pull/61/commits/4b86fef7d2411faf48dec10cc8417e80d0551943) 
Changes: 
- Changed variable names
- added Bible renderer to support download.json and whole.json files
- Sorry to add another feature while in the middle of this PR, but I added one other property (bytecount) to the chapters model and to the download index.  Even though Cloudflare is in front of the production blobs and thus will delievered a compressed response, The motivation was so that any end consumer could ultimately know how much data would have to be transferred and uncompressed.  Someone might be less inclined to try to download an entire 10mb ulb on bad connections, and instead save it for later on better connections for example.

rbnswartz commented on ScriptureRenderingPipeline/Renderers/BibleRenderer.cs:
```diff
@@ -16,152 +16,177 @@
+				// Since the print all page isn't going to broken up then just write stuff out here
+				printBuilder.AppendLine(content);
+			}
+			long totalByteCount = downloadIndex.Content
```
Linq is powerful but it can end up being a performance problem if you aren't careful
